internal/controller/components/dashboard/dashboard_support_test.go

```diff
@@ -0,0 +1,712 @@
+package dashboard_test
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	routev1 "github.com/openshift/api/route/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/api/common"
+	componentApi "github.com/opendatahub-io/opendatahub-operator/v2/api/components/v1alpha1"
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/api/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/internal/controller/components/dashboard"
+	"github.com/opendatahub-io/opendatahub-operator/v2/internal/controller/status"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/utils/test/fakeclient"
+
+	. "github.com/onsi/gomega"
+)
+
+const (
+	dashboardURLKey = "dashboard-url"
+	sectionTitleKey = "section-title"
+	testNamespace   = "test-namespace"
+	testDashboard   = "test-dashboard"
+	testPath        = "/test/path"
+
+	// Test condition types - these should match the ones in dashboard_support.go.
+	testConditionTypes = status.ConditionDeploymentsAvailable
+)
+
+func TestDefaultManifestInfo(t *testing.T) {
+	tests := []struct {
+		name     string
+		platform common.Platform
+		expected string
+	}{
+		{
+			name:     "OpenDataHub platform",
+			platform: cluster.OpenDataHub,
+			expected: "/odh",
+		},
+		{
+			name:     "SelfManagedRhoai platform",
+			platform: cluster.SelfManagedRhoai,
+			expected: "/rhoai/onprem",
+		},
+		{
+			name:     "ManagedRhoai platform",
+			platform: cluster.ManagedRhoai,
+			expected: "/rhoai/addon",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+			manifestInfo := dashboard.DefaultManifestInfo(tt.platform)
+			g.Expect(manifestInfo.ContextDir).Should(Equal(dashboard.ComponentName))
+			g.Expect(manifestInfo.SourcePath).Should(Equal(tt.expected))
+		})
+	}
+}
+
+func TestBffManifestsPath(t *testing.T) {
+	g := NewWithT(t)
+	manifestInfo := dashboard.BffManifestsPath()
+	g.Expect(manifestInfo.ContextDir).Should(Equal(dashboard.ComponentName))
+	g.Expect(manifestInfo.SourcePath).Should(Equal("modular-architecture"))
```

Comment:
Should this be a constant in code?

Reply:
Yes, will add a constant ModularArchitectureSourcePath = "modular-architecture" on dashboard_support.go and update BffManifestsPath() to use it